pydm/widgets/base.py

```diff
@@ -179,6 +180,8 @@ def value_changed(self, new_val):
         """
         self.value = new_val
         self.channeltype = type(self.value)
+        if self.channeltype == np.ndarray:
+            self.subtype = type(self.value[0])
```

Comment:
Would you mind changing to the following:
```python
self.subtype = self.value.dtype.type
```
This will avoid a `IndexError` in case of a zero length array (weird but it can happen)